### PR TITLE
unit test for kubectl config set

### PR DESCRIPTION
**Pull request by zjj2wry**
**What this PR does / why we need it**:
unit test for kubectl config set

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
think you

**Release note**:

```release-note
```

### CODE REVIEW

**Comment by zjj2wry** [comment]
@deads2k

File: pkg/kubectl/cmd/config/set_test.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+	if err != nil {
+		t.Fatalf("unexpected error loading kubeconfig file: %v", err)
+	}
+	if len(test.expected) != 0 {
+		if buf.String() != test.expected {
+			t.Errorf("Failded in:%q\n expected %v\n but got %v", test.expected, buf.String())
+		}
+	}
+	if len(test.args) == 2 {
+		if test.args[1] != config.CurrentContext {
```
Comment (deads2k) [inline pkg/kubectl/cmd/config/set_test.go]: This means that the test struct only works for testing set against the current context right?  That's ok, but its more restrictive than the name suggests.  How about comparing against the `expectedConfig` instead?

Comment (zjj2wry) [inline pkg/kubectl/cmd/config/set_test.go]: yes,i start  want compare config , but  real config will initialize some nil map , and LocationOfOrigin etc. so my expectConfig can not initialize completely in testcases.

Comment (deads2k) [inline pkg/kubectl/cmd/config/set_test.go]: > yes,i start want compare config , but real config will initialize some nil map , and LocationOfOrigin etc. so my expectConfig can not initialize completely in testcases.

Before the comparison, you could clear the `LocationOfOrigin` and you could initialize the maps in your `expectConfig`.

Comment (zjj2wry) [inline pkg/kubectl/cmd/config/set_test.go]: sorry to reply late. i will update like you say. think you very much

**Comment by zjj2wry** [comment]
@k8s-bot ok to test

**Comment by deads2k** [comment]
@k8s-bot  ok to test

**Comment by zjj2wry** [comment]
@k8s-bot verify test this

**Comment by deads2k** [comment]
@k8s-bot verify test this

**Pull request merged**